apps/server/lib/lexical/server/project/diagnostics.ex

```diff
@@ -137,11 +137,23 @@ defmodule Lexical.Server.Project.Diagnostics do
     defp position_to_range(%SourceFile{} = source_file, {line_number, column}) do
       line_number = Math.clamp(line_number, 1, SourceFile.size(source_file))
       column = max(column, 1)
+      to_lsp_range(line_number, column, source_file)
+    end
+
+    defp position_to_range(source_file, line) when is_integer(line) do
+      line_number = Math.clamp(line, 1, SourceFile.size(source_file))
 
+      with {:ok, line_text} <- SourceFile.fetch_text_at(source_file, line_number) do
+        column = Math.count_leading_spaces(line_text) + 1
+        to_lsp_range(line_number, column, source_file)
+      end
+    end
+
+    defp to_lsp_range(line, column, source_file) do
       elixir_range =
         ExRange.new(
-          ExPosition.new(line_number, column),
-          ExPosition.new(line_number + 1, 1)
+          ExPosition.new(line, column),
+          ExPosition.new(line + 1, 1)
```

Comment:
line_number is more descriptive; In this codebase, the word `line` refers to a specific data structure.